lib/tokenize.cpp

```diff
@@ -10114,8 +10114,12 @@ void Tokenizer::simplifyKeyword()
             tok->deleteNext();
 
         if (c99) {
-            while (tok->str() == "restrict")
-                tok->deleteThis();
+            if (tok->str() == "restrict") {
+                // check if restrict is used as a type qualifier
+                if (Token::Match(tok->previous(), "*|[ restrict %name%") ||
+                    Token::Match(tok->next(), "* %name%"))
```

Comment:
then we have such code also:

    void foo(int restrict) { a = buf[restrict * restrict]; }